in_toto/model.go

```diff
@@ -57,6 +79,42 @@ type Link struct {
 	Environment map[string]interface{} `json:"environment"`
 }
 
+func validateMaterialOrProductHash(value string) error {
+	hashSchemaCheck, _ := regexp.MatchString("^[a-fA-F0-9]+$", value)
+	if !hashSchemaCheck {
+		return fmt.Errorf("hash value has invalid format")
+	}
+	return nil
+}
+
+func validateLink(link Link) error {
+	if link.Type != "link" {
+		return fmt.Errorf("invalid type for link: should be 'link'")
+	}
+
+	for _, material := range link.Materials {
```

Comment:
... these two loops could be generalized into a single `validateArtifact` function. What do you think?